tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/util/TypeUtils.java

```diff
@@ -121,7 +121,14 @@ public static String extractClosestRealTypeAsString(TypeMirror type, Context con
 			return extractClosestRealTypeAsString( compositeUpperBound, context );
 		}
 		else {
-			return context.getTypeUtils().erasure( type ).toString();
+			final TypeMirror erasureType = context.getTypeUtils().erasure( type );
+			if ( TypeKind.ARRAY.equals( erasureType.getKind() ) ) {
+				// keep old behavior here for arrays since #asElement returns null for them.
+				return erasureType.toString();
+			}
+			else {
+				return context.getTypeUtils().asElement( erasureType ).toString();
```

Comment:
@Naros why use `toString()` instead of `getQualifiedName()`? I would have expected the latter to return what we want in a more future-proof way.

Reply:
@gsmet When I push the test for embeddables for our metamodel test coverage, I'll include that too.